ZenPacks/zenoss/HttpMonitor/http.py

```diff
@@ -164,13 +168,42 @@ def _getBody(self, response):
         self._response = response
         return readBody(response).addErrback(self._pageErr)
 
-    def _answer(self, body):
+    def _answer(self, body=""):
         res = dict()
-        body = body if not self._body else self._body
+        body = str(body if not self._body else self._body)
         res['body'] = body
         res['headers'] = self._response.headers
         res['code'] = self._response.code
         res['message'] = self._response.phrase
         res['time'] = time.time() - self._startTime
-        res['size'] = self._bodysize(body)
+        res['size'] = self._bodysize(body+str(res['headers']))
+        if self._regex:
+            regex = self._checkRegex(body)
+            if regex and regex.get('status', ""):
+                res['msg'] = regex
         return res
+
+    def regex(self, regex, caseSensitive=False, invert=False):
+        self._regex = regex
+        self._caseSensitive = caseSensitive
+        self._invert = invert
+
+    def _checkRegex(self, body=""):
+        import re
+        try:
+            if self._caseSensitive:
+                regex = re.compile(self._regex)
+            else:
+                regex = re.compile(self._regex, re.IGNORECASE)
+        except re.error as err:
+            return {'status': 'CRITICAL', 'msg': 'Could not compile regular expression: '+
+                    repr(self._regex)}
+
+        match = bool(regex.search(body))
+        if (match and self._invert) or (not match and self._invert):
```

Comment:
logic is wrong here.  this will `pass` (ie. return None) if self._invert is True, regardless of match state